framework/test/unit/modules/random/endpoint.spec.ts

```diff
@@ -148,10 +158,265 @@ describe('RandomModuleEndpoint', () => {
+
+			// Act & Assert
+			await expect(randomEndpoint.setHashOnion(context)).rejects.toThrow(
+				"Lisk validator found 1 error[s]:\nProperty '.seed' should be of type 'string'",
+			);
+		});
+
+		it('should throw error when count provided in params is invalid', async () => {
+			// Arrange
+			const { address } = genesisDelegates.delegates[0];
+			const seed = genesisDelegates.delegates[0].hashOnion.hashes[1];
+			const count = 'count';
+			const distance = 1000;
+			context.params = { address, seed, count, distance };
+
+			// Act & Assert
+			await expect(randomEndpoint.setHashOnion(context)).rejects.toThrow(
+				"Lisk validator found 1 error[s]:\nProperty '.count' should be of type 'integer'",
+			);
+		});
+
+		it('should throw error when distance provided in params is invalid', async () => {
+			// Arrange
+			const { address } = genesisDelegates.delegates[0];
+			const seed = genesisDelegates.delegates[0].hashOnion.hashes[1];
+			const count = 1000;
+			const distance = 'distance';
+			context.params = { address, seed, count, distance };
+
+			// Act & Assert
+			await expect(randomEndpoint.setHashOnion(context)).rejects.toThrow(
+				"Lisk validator found 1 error[s]:\nProperty '.distance' should be of type 'integer'",
+			);
+		});
+
+		it('should throw error when count is less than 1', async () => {
+			// Arrange
+			const { address } = genesisDelegates.delegates[0];
+			const seed = genesisDelegates.delegates[0].hashOnion.hashes[1];
+			const count = 0;
+			const distance = 1000;
+			context.params = { address, seed, count, distance };
+
+			// Act & Assert
+			await expect(randomEndpoint.setHashOnion(context)).rejects.toThrow(
+				'Lisk validator found 1 error[s]:\nmust be >= 1',
+			);
+		});
+
+		it('should throw error when distance is less than 1', async () => {
+			// Arrange
+			const { address } = genesisDelegates.delegates[0];
+			const seed = genesisDelegates.delegates[0].hashOnion.hashes[1];
+			const count = 1000;
+			const distance = 0;
+			context.params = { address, seed, count, distance };
+
+			// Act & Assert
+			await expect(randomEndpoint.setHashOnion(context)).rejects.toThrow(
+				'Lisk validator found 1 error[s]:\nmust be >= 1',
+			);
+		});
+	});
+
+	describe('getSeeds', () => {
+		let address: string;
+
+		beforeEach(async () => {
+			// Arrange
+			address = genesisDelegates.delegates[0].address;
+			const seed = genesisDelegates.delegates[0].hashOnion.hashes[1];
+			const count = 1000;
+			const distance = 10;
+
+			await randomEndpoint.setHashOnion({ ...context, params: { address, seed, count, distance } });
+		});
+
+		it('should return an array of seed objects', async () => {
+			// Act
+			const storedSeed = await randomEndpoint.getHashOnionSeeds(context);
+
+			// Assert
+			expect(storedSeed.seeds).toHaveLength(1);
+			expect(storedSeed.seeds[0]).toEqual({
+				address,
+				count: 1000,
+				distance: 10,
+				seed: expect.any(String),
+			});
+		});
+	});
+
+	describe('hasSeed', () => {
```

Comment:
would be nice to have a test case for empty input params i.e. `params: { }`

same for other tests